contentcuration/contentcuration/views/admin.py

```diff
@@ -149,16 +157,192 @@ def get_channel_kind_count(request, channel_id):
+    user_serializer = SimpleAdminUserListSerializer(user_list, many=True)
 
     return Response(user_serializer.data)
 
 
+class ChannelUserListPagination(PageNumberPagination):
+    page_size = DEFAULT_ADMIN_PAGE_SIZE
+    page_size_query_param = 'page_size'
+    max_page_size = 500
+
+    def get_paginated_response(self, data):
+        return Response({
+            'links': {
+                'next': self.get_next_link(),
+                'previous': self.get_previous_link()
+            },
+            'count': self.page.paginator.count,
+            'total_pages': self.page.paginator.num_pages,
+            'results': data
+        })
+
+
+class AdminChannelListFilter(django_filters.FilterSet):
+    published = django_filters.BooleanFilter(
+        name='main_tree__published',
+    )
+    staged = django_filters.BooleanFilter(
+        name='staging_tree'
+    )
+    ricecooker_version__isnull = django_filters.rest_framework.BooleanFilter(
+        name='ricecooker_version',
+        lookup_expr='isnull'
+    )
+
+    class Meta:
+        model = Channel
+        fields = (
+            'name',
+            'id',
+            'editors__id',
+            'deleted',
+            'public',
+            'staging_tree',
+            'staged',
+            'ricecooker_version',
+            'deleted',
+            'published'
+        )
+
+
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework.filters import OrderingFilter
+from rest_framework.filters import SearchFilter
+
+
+class AdminChannelListView(generics.ListAPIView):
+    serializer_class = AdminChannelListSerializer
+    filter_backends = (DjangoFilterBackend, OrderingFilter, SearchFilter)
+    filter_class = AdminChannelListFilter
+    pagination_class = ChannelUserListPagination
+    authentication_classes = (SessionAuthentication, BasicAuthentication, TokenAuthentication,)
+    permission_classes = (IsAdminUser,)
+
+    search_fields = (
+        'name',
+        '=id',
+        'editors__first_name',
+        'editors__last_name',
+        '=editors__email',
+    )
+
+    ordering_fields = (
+        'name',
+        'id',
+        'priority',
+        'users',
+        'items',      # not working yet!
+        'modified',
+        'created',
+    )
+    ordering = ('name',)
+    
+    def get_queryset(self):
+
+        # (This part requires django 1.11, and isn't quite working!)
+        # from django.db.models import OuterRef
+        # from django.db.models.functions import Cast
+        # from django.db.models.functions import Coalesce
+        # from django.db.models import Subquery
+        # from django.db.models import Int
+
+        # modified = ContentNode.objects\
+        #             .filter(tree_id=OuterRef('main_tree__tree_id'))\
+        #             .order_by()\
+        #             .values('tree_id')\
+        #             .annotate(m=Max('modified'))\
+        #             .values('m')
+                        
```

Comment:
I had trouble getting this stuff to work ^^^

Reply:
No worries! We can peer program on it later if we need to